frontend/src/app/Catalog/CatalogView/CatalogView.tsx

```diff
@@ -130,11 +256,115 @@ const CatalogView = (props: CatalogViewProps) => {
 		});
 	};
 
-	const enrollPath = legacyId
-		? `/enrollment/0-${legacyId}-${semester.replace(' ', '-')}-all`
-		: `/enrollment`;
+	
 
-	const gradePath = legacyId ? `/grades/0-${legacyId}-all-all` : `/grades`;
+
+	const [hoveredClass, setHoveredClass] = useState<any>(false);
+	const [updateMobileHover, setUpdateMobileHover] = useState<any>(true);
+	
+
+	function update(course:any, grade:any) {
+		const gradesData = (store.getState() as any).grade.gradesData;
+		if (course && gradesData && gradesData.length > 0) {
+			const selectedGrades = gradesData.filter((c:any) => course.id === c.id)[0];
+			const hoverTotal = {
+				...course,
+				...selectedGrades,
+				hoverGrade: grade
+			};
+
+			setHoveredClass(hoverTotal);
+		}
+	}
+
+	// Handler function for updating GradesInfoCard on hover with single course
+	function updateGraphHover(data:any) {
+		const { isTooltipActive, activeLabel } = data;
+		const selectedCourses = (store.getState() as any).grade.selectedCourses;
+
+		const noBarMobile = updateMobileHover && window.innerWidth < 768;
+
+		// Update the selected course if no bar is clicked if in mobile
+		if (isTooltipActive && (selectedCourses.length === 1 || noBarMobile)) {
+			const selectedCourse = selectedCourses[0];
+			const grade = activeLabel;
+			update(selectedCourse, grade);
+		}
+
+		// Update mobile hover records if there actually is a bar (then we only want updateBarHover to run)
+		setUpdateMobileHover({ updateMobileHover: true });
+	}
+
+	const [tab, setTab] = useState<Number>(0)
+	const openTab = () => {
```

Comment:
Since the `CatalogView` is now 500 lines and the tabs derive it's own state and functions, lets refactor it into its own component, as it has functionality that the `CatalogView` doesn't need to know about.